lib/monad.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// id :: a -> a
+function id(x) {
+  return x;
+}
+
+// curry :: ((a, b) -> c) -> a -> b -> c
+function curry(f, ...args1) {
+  return (...args2) => f(...args1, ...args2);
+}
+
+function withCb(func) {
```

Comment:
Closure is needed because I want to separate giving `func` and giving `...args` and `callback`. I give `func` and user gives `...args` and `callback`.
`func` gets arguments as array at first argument and callback at second. 
`withCb` solves problem of splitting arguments and callback for functions with vary parameters count. It would be easy if javascript would allowed such definition `(...args, callback) => {}`. But it didn't so for avoiding repeating of those lines:
```javascript
const callback = argsCb[argsCb.length - 1];
const args = argsCb.slice(0, argsCb.length - 1);
```
func(args, callback) can be wrapped in a such way that result function is func(...args, callback).